### PR TITLE
Actually unique `RowId`s 1: Forbid use of duplicate `RowId`s

### PR DESCRIPTION
**Pull request by teh-cmc**
In this PR:
- Detect and forbid duplicated `RowId`s in the store
- Update test suites accordingly

---

_Part of a small PR series whose goal is to make `RowId`s truly unique: i.e. a `RowId` corresponds to one insertion in the store, using the store's atomic unit of insertion -- the `DataRow` (which itself might still carry arbitrarily many timestamps(!)).
This creates an invariant that downstream subscribers of store events can trust and build on: a RowId is introduced once and GC'd once, that's it._

- #4174 
- #4175 
- #4176
- #4180

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4174) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4174)
- [Docs preview](https://rerun.io/preview/8a88d0ab863a6ae073a87b2f32ab6b6efa5e5d8e/docs) 
- [Examples preview](https://rerun.io/preview/8a88d0ab863a6ae073a87b2f32ab6b6efa5e5d8e/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by teh-cmc** [comment]
Uh-oh, I broke something.

**Review by Wumpf — APPROVED** [review]
lgtm!

**Review by Wumpf — APPROVED** [review]
lgtm!

**Comment by Wumpf** [comment]
curious, almost all our tests are fine with this (after slight fixup); so we never actually hit RowId duplication in the wild so far..?

File: crates/re_arrow_store/src/store_write.rs [hunk 1]
```diff
@@ -234,38 +237,20 @@ impl DataStore {
...
-                        }
-                    } else {
-                        // NOTE: In a map, thus on the heap!
-                        added_size_bytes += timeline.total_size_bytes();
-                        added_size_bytes += time.as_i64().total_size_bytes();
-                    }
-                }
-            }
-        }
+        self.insert(row_id, timepoint);
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_arrow_store/src/store_write.rs:249]: I just find the early-out so much more readable, but you're right... :(

File: crates/re_log_types/src/lib.rs [hunk 2]
```diff
@@ -21,7 +21,6 @@ pub mod arrow_msg;
 mod data_cell;
 mod data_row;
 mod data_table;
-#[cfg(feature = "testing")]
 pub mod example_components;
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_log_types/src/lib.rs:24]: Oh, I thought I was the one who did.

It cost me a lot of time hunting down why something was working on CI vs. not on my machine and that was due to usual hidden-additive-workspace-features as usual...
But maybe I can add it back now actually.

**Comment by teh-cmc** [comment]
> curious, almost all our tests are fine with this (after slight fixup); so we never actually hit RowId duplication in the wild so far..?
> 
> EDIT: Ah. It happens when you have several processes logging to the same recording, right?

Oh no on the contrary, almost all our tests failed with this ^_^ (which is expected, they just copy rows left and right).
The slight fix up is really not so slight and makes it so every test just tries to increment RowId for as long as necessary until it's finally valid.

**Pull request merged**